### PR TITLE
Allow bridging if cast window is active

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description

- Before we only allowed bridging during the L2 internal voting period, but instead we should allow vote bridging during the cast window

### CODE REVIEW

**Comment by alexkeating** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting/95?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #93** <a href="https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting/93?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #95** <a href="https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting/95?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ScopeLift/l2-flexible-voting/95?utm_source=stack-comment).

File: src/L2GovernorMetadata.sol [hunk 1]
```diff
@@ -36,7 +36,7 @@ abstract contract L2GovernorMetadata {
   // before the end of the proposal to cast the vote.
   /// @dev If the block conversion hack is removed from this contract, then this storage var is
   /// probably not needed in this contract and can probably be moved back to the `L2VoteAggregator`
-  uint32 public constant CAST_VOTE_WINDOW = 1200;
+  uint32 public CAST_VOTE_WINDOW = 1200;
```
Comment (wildmolasses) [inline src/L2GovernorMetadata.sol:39]:
```suggestion
  uint32 public immutable CAST_VOTE_WINDOW = 1200;
```

File: src/WormholeL2GovernorMetadata.sol [hunk 2]
```diff
@@ -9,10 +9,10 @@ import {L2GovernorMetadata} from "src/L2GovernorMetadata.sol";
 contract WormholeL2GovernorMetadata is L2GovernorMetadata, WormholeReceiver {
   /// @param _relayer The address of the WormholeL2GovernorMetadata contract.
   /// @param _owner The address that will become the contract owner.
-  constructor(address _relayer, address _owner, address _l1BlockAddress)
+  constructor(address _relayer, address _owner, address _l1BlockAddress, uint32 castWindow)
```
Comment (wildmolasses) [inline src/WormholeL2GovernorMetadata.sol:12]: nit

```suggestion
  constructor(address _relayer, address _owner, address _l1BlockAddress, uint32 _castWindow)
```

File: src/optimized/WormholeL2GovernorMetadataOptimized.sol [hunk 3]
```diff
@@ -11,8 +11,8 @@ contract WormholeL2GovernorMetadataOptimized is WormholeL2GovernorMetadata {
   /// @notice The ID of the proposal mapped to an internal proposal ID.
   mapping(uint256 governorProposalId => uint16) public optimizedProposalIds;
 
-  constructor(address _relayer, address _owner, address _l1BlockAddress)
-    WormholeL2GovernorMetadata(_relayer, _owner, _l1BlockAddress)
+  constructor(address _relayer, address _owner, address _l1BlockAddress, uint32 castWindow)
```
Comment (wildmolasses) [inline src/optimized/WormholeL2GovernorMetadataOptimized.sol:14]: nit

```suggestion
  constructor(address _relayer, address _owner, address _l1BlockAddress, uint32 _castWindow)
```

File: test/L2VoteAggregator.t.sol [hunk 4]
```diff
@@ -384,7 +389,8 @@ contract CastVoteWithReason is L2VoteAggregatorTest {
 
     l2Erc20.mint(address(this), _amount);
 
-    L2GovernorMetadata.Proposal memory l2Proposal = voteAggregator.createProposal(1, 3000, false);
+    L2GovernorMetadata.Proposal memory l2Proposal =
+      voteAggregator.createProposal(1, uint128(voteAggregator.CAST_VOTE_WINDOW()) + 1, false);
```
Comment (wildmolasses) [inline test/L2VoteAggregator.t.sol:393]: nit: these tests all extend from `L2VoteAggregatorTest`. Can we define a variable there called `DEFAULT_VOTE_END` and set it to `uint128(voteAggregator.CAST_VOTE_WINDOW()) + 1` and use it in all these places?

```suggestion
      voteAggregator.createProposal(1, DEFAULT_VOTE_END, false);
```

File: test/WormholeL1VotePool.t.sol [hunk 5]
```diff
@@ -96,7 +97,7 @@ contract _ReceiveCastVoteWormholeMessages is WormholeL1VotePoolTest {
     vm.deal(address(this), 10 ether);
 
     l2VoteAggregator.createProposalVote(_proposalId, _l2Against, _l2For, _l2Abstain);
-    l2VoteAggregator.createProposal(_proposalId, 3000);
+    l2VoteAggregator.createProposal(_proposalId, uint128(l2VoteAggregator.CAST_VOTE_WINDOW()) + 1);
```
Comment (wildmolasses) [inline test/WormholeL1VotePool.t.sol:100]: nit: same thing here, would prefer extracting to a var

```suggestion
    l2VoteAggregator.createProposal(_proposalId, DEFAULT_VOTE_END);
```

**Review by wildmolasses — COMMENTED** [review]
a few nits, thanks @alexkeating

**Review by wildmolasses — APPROVED** [review]

**Pull request merged**